omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/SmartOnFhirRestController.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.web.controller;
+
+import javax.crypto.spec.SecretKeySpec;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.crypto.JavaAlgorithm;
+import org.keycloak.crypto.KeyWrapper;
+import org.keycloak.crypto.MacSignatureSignerContext;
+import org.keycloak.crypto.SignatureSignerContext;
+import org.keycloak.jose.jws.JWSBuilder;
+import org.keycloak.representations.JsonWebToken;
+import org.openmrs.module.smartonfhir.util.SmartSecretKeyHolder;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@RequestMapping("/rest/" + RestConstants.VERSION_1 + SmartOnFhirRestController.SMARTONFHIR_NAMESPACE)
+public class SmartOnFhirRestController extends BaseRestController {
+	
+	public static final String SMARTONFHIR_NAMESPACE = "/smartonfhir";
+	
+	@RequestMapping(value = "/accessConformation", method = RequestMethod.GET)
+	public void accessConformation(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		
+		String token = request.getParameter("token");
+		
+		if (token == null) {
+			response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+			return;
+		}
+		
+		JsonWebToken tokenSendBack = new JsonWebToken();
+		tokenSendBack.setOtherClaims("username", "admin");
```

Comment:
I realise this is just a dummy implementation, but the code to get the current user name is something like:

```java
User user = Context.getAuthenticatedUser();
if (user == null) {
    response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
    return;
}

tokenSendBack.setSubject(user.getUsername());
```

Note the use of `setSubject()`. Since we're using a [`JsonWebToken`](https://www.jsonwebtoken.io/), it makes sense to use the relevant fields when they're provided.